File: 101/exercise_rework/easy_6/right_triangles.rb
# right_triangles.rb

def produce_tri(n)
  right_tri = ''
  n.times do |idx|
    spaces = n - idx
    right_tri << "#{' ' * spaces}#{'*' * idx}\n"
  end
  right_tri
end

def triangle(n)
  puts produce_tri(n)
end

File: 210/lesson_3/logic_and_flow/small_problems/easy2/right_triangles.js
// right_triangles.js

function multiplyCharacter(char, times) {
  var repeatedChars = '';
  for (var i = 0; i < times; i += 1) {
    repeatedChars += char;
  }
  return repeatedChars;
}

function triangle(limit) {
  var spaces;
  var asterisks;
  var i = 0;

  while (i < limit) {
    i += 1;
    spaces = multiplyCharacter(' ', limit - i);
    asterisks = multiplyCharacter('*', i);
    console.log(spaces + asterisks);
  }
}

triangle(5);
triangle(9);